### ISSUE TITLE
search function and stopwords

### ISSUE THREAD

**Issue by stcoats** [huggingface/datasets-server#3104]
It seems that the dataset-viewer search function returns no hits if one searches for terms such as “what”, “can”, “which”, and so on. Has the indexing function removed stopwords like this? The rows are returned if one uses the SQL console, but the returned rows in the SQL console don’t give access to the column with audio, for a dataset that includes audio files. Is there a way to search for stop words like this in the default datasets Viewer? It would be really useful if all of the textual content in a column could be searchable.

**Comment by AndreaFrancis** [comment]
> Has the indexing function removed stopwords like this? 

Yes, we use a default list of stopwords, which contains 571 words, including "what," "can," and "which." You can view the complete list here:
[DuckDB English Stopwords List](https://github.com/duckdb/duckdb/blob/4982c3f135d0f55cbb896ca35a0cd8dc458fc4ee/extension/fts/fts_indexing.cpp#L58).

But, as @severo  mentioned in [this discussion](https://discuss.huggingface.co/t/datasets-viewer-searching-for-text/122560/3), we now support [language-specific stemmers](https://github.com/huggingface/dataset-viewer/issues/1437) for monolingual datasets. Using a default English stopwords list for all languages no longer makes sense. However, DuckDB currently lacks a straightforward way to assign stopwords based on language as it does for stemmers (we would need to seed a stopwords table for non-English datasets). Therefore, for now, the best approach is to set the stopwords parameter to 'none'. cc. @lhoestq 

If users want to remove stopwords for specific monolingual datasets (e.g., English), this could be a candidate for a [custom configuration](https://github.com/huggingface/dataset-viewer/issues/2639) at the dataset card level. Keep in mind that removing stopwords like "what," "can," or "which" helps focus on more meaningful terms, improving search relevance. It also reduces the size of the search index and speeds up queries, which is crucial for performance in the Datasets Viewer.

**Comment by stcoats** [comment]
@AndreaFrancis thanks for the reply. It is true that content words are typically more meaningful than are prepositions, articles, and so on, but it is very much the case that a researcher may be interested in filtering a dataset for function words or, especially, collocations containing function words. The way the plugin works at the moment, a search string like "never experience" is equivalent to "experience". 
In addition, and as you note, the current configuration actually removes words in datasets in other languages if they are homographs with English stopwords. So, for example Swedish "by" (village) is unsearchable in a Swedish-language dataset with the current viewer parameters. 
I understand that it affects speed, but I think it would be great if no stopwords are used and/or custom configurations could be made available.

**Comment by cfahlgren1** [comment]
> The rows are returned if one uses the SQL console, but the returned rows in the SQL console don’t give access to the column with audio, for a dataset that includes audio files.

Hi @stcoats we just shipped Audio / Image support in SQL Console Results. I am not sure which dataset you are trying to view / query, but it should work now 🤗 I know it doesn't fix the stemming, but hopefully can still help! Let me know if you have any questions


![image](https://github.com/user-attachments/assets/039c0aed-3eca-4f67-b51f-d461474309ac)
